File: rockpaperscissors/src/main/java/RockPaperScissorsGame.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class RockPaperScissorsGame extends ThrowingGame {
    private final List<Gesture> winningGestureDecider = new ArrayList<Gesture>(Arrays.asList(
            Gesture.ROCK,
            Gesture.SCISSORS,
            Gesture.PAPER)
    );

    RockPaperScissorsGame(Player player, Player bot) {
        super(player, bot);
    }

    Gesture getWinningGesture(Gesture player1Gesture, Gesture player2Gesture) {
        if(player1Gesture.equals(player2Gesture))
            return Gesture.DRAW;
        winningGestureDecider.remove(player1Gesture);
        winningGestureDecider.remove(player2Gesture);
        Gesture remainingGesture = winningGestureDecider.get(0);
        switch (remainingGesture){
            case ROCK: return Gesture.SCISSORS;
            case PAPER: return Gesture.ROCK;
            case SCISSORS: return Gesture.PAPER;
        }
        return Gesture.DRAW;
    }
}

File: rockpaperscissors_py/main/RockPaperScissorsGame.py
from main import GestureEnum as Gesture

class RockPaperScissorsGame:
    def __init__(self, player1, player2):
        self.player1 = player1
        self.player2 = player2

    def determine_result(self, gesture_thrown_by_player1, gesture_thrown_by_player2):
        if gesture_thrown_by_player1 == gesture_thrown_by_player2:
            return "DRAW"
        gesture_list = [Gesture.ROCK, Gesture.PAPER, Gesture.SCISSORS]
        gesture_list.remove(gesture_thrown_by_player1)
        gesture_list.remove(gesture_thrown_by_player2)
        winning_gesture = self.get_winning_gesture(gesture_list[0])
        if gesture_thrown_by_player1 == winning_gesture:
            return self.player1.get_name()
        return self.player2.get_name()

    def get_winning_gesture(self, gesture):
        return {
            Gesture.ROCK: Gesture.SCISSORS,
            Gesture.PAPER: Gesture.ROCK,
            Gesture.SCISSORS: Gesture.PAPER
        }.get(gesture, Gesture.DRAW)
